src/Container.js

```diff
@@ -45,13 +47,15 @@ export default class Container {
         child._resolvers = Object.create(this._resolvers);
         child._isDisposed = false;
         child._childContainers = [];
+        child._registerSelf();
```

Comment:
I don't think so. This aids resolution of the correct containers for objects being resolved. There should be no side effects of doing this and all it really does is put a 'built in' register  into the container. I think it's important to be a default.